### PR TITLE
Add kernel profiler for oneapi interfaces

### PR DESCRIPTION
**Pull request by dmitrii-kriukov**

### CODE REVIEW

File: cpp/oneapi/dal/detail/profiler.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+class profiler_task {
+public:
+    profiler_task(const char* taskName);
+#ifdef ONEDAL_DATA_PARALLEL
+    profiler_task(const char* taskName, sycl::queue& q);
+#endif
+    ~profiler_task();
+
+private:
+    const char* _taskName;
```
Comment (vlad-nazarov) [inline cpp/oneapi/dal/detail/profiler.hpp:51]:
```suggestion
    const char* task_name_;
```

File: cpp/oneapi/dal/detail/profiler.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ONEDAL_PROFILER_GET_MACRO(__VA_ARGS__,                                              \
+                                  ONEDAL_PROFILER_MACRO_2,                                  \
+                                  ONEDAL_PROFILER_MACRO_1,                                  \
+                                  FICTIVE)(__VA_ARGS__)
+
+namespace oneapi::dal::detail {
+
+class profiler_task {
+public:
+    profiler_task(const char* taskName);
```
Comment (vlad-nazarov) [inline cpp/oneapi/dal/detail/profiler.hpp:44]:
```suggestion
    profiler_task(const char* task_name);
```

File: cpp/oneapi/dal/detail/profiler.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+#ifdef ONEDAL_DATA_PARALLEL
+    profiler_task(const char* taskName, sycl::queue& q);
+#endif
+    ~profiler_task();
+
+private:
+    const char* _taskName;
+#ifdef ONEDAL_DATA_PARALLEL
+    sycl::queue task_queue;
```
Comment (vlad-nazarov) [inline cpp/oneapi/dal/detail/profiler.hpp:53]:
```suggestion
    sycl::queue task_queue_;
```

File: cpp/oneapi/dal/detail/profiler.hpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                                  ONEDAL_PROFILER_MACRO_1,                                  \
+                                  FICTIVE)(__VA_ARGS__)
+
+namespace oneapi::dal::detail {
+
+class profiler_task {
+public:
+    profiler_task(const char* task_name);
+#ifdef ONEDAL_DATA_PARALLEL
+    profiler_task(const char* task_name, sycl::queue& task_queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/profiler.hpp:46]: const sycl::queue& task_queue?

File: cpp/oneapi/dal/detail/profiler.hpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#ifdef ONEDAL_DATA_PARALLEL
+    sycl::queue task_queue_;
+#endif
+};
+
+class profiler {
+public:
+    static profiler_task start_task(const char* task_name);
+#ifdef ONEDAL_DATA_PARALLEL
+    static profiler_task start_task(const char* task_name, sycl::queue& task_queue);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/profiler.hpp:61]: const sycl::queue& task_queue?

File: cpp/oneapi/dal/detail/profiler.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+profiler_task profiler::start_task(const char* task_name) {
+    return profiler_task(task_name);
+}
+
+void profiler::end_task(const char* task_name) {}
+
+profiler_task::profiler_task(const char* task_name) : task_name_(task_name) {}
+
+#ifdef ONEDAL_DATA_PARALLEL
+profiler_task profiler::start_task(const char* task_name, sycl::queue& task_queue) {
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/profiler.cpp:29]: const sycl::queue& task_queue?

File: cpp/oneapi/dal/detail/profiler.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    return profiler_task(task_name, task_queue);
+}
+
+profiler_task::profiler_task(const char* task_name, sycl::queue& task_queue)
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/profiler.cpp:33]: const sycl::queue& task_queue?

**Review by amgrigoriev — APPROVED** [review]

**Review by KulikovNikita — APPROVED** [review]

**Review by vlad-nazarov — APPROVED** [review]

**Pull request merged**